nntrainer/models/dynamic_training_optimization.h

```diff
@@ -0,0 +1,183 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2020 Parichay Kapoor <pk.kapoor@samsung.com>
+ *
+ * @file   dynamic_training_optimization.h
+ * @date   4 January 2021
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Parichay Kapoor <pk.kapoor@samsung.com>
+ * @bug    No known bugs except for NYI items
+ * @brief  This is Dynamic Training Optimization for Neural Network
+ *
+ */
+
+#ifndef __DYNAMIC_TRAINING_OPT_H__
+#define __DYNAMIC_TRAINING_OPT_H__
+#ifdef __cplusplus
+
+#include <random>
+#include <vector>
+
+#include <layer_internal.h>
+#include <tensor.h>
+
+namespace nntrainer {
+
+/**
+ * @class   DynamicTraining Optimization
+ * @brief   Dynamic Training Optimization
+ */
+class DynamicTrainingOptimization {
+public:
+  /**
+   * @brief     Constructor of DynamicFineTuning Optimization
+   */
+  DynamicTrainingOptimization(int threshold_ = 1, int skip_n_iter = 1);
+
+  /**
+   * @brief     Set threshold for optimization
+   */
+  void setThreshold(float threshold_) { threshold = threshold_; };
+
+  /**
+   * @brief     Set the reduce operation for dynamic optimization
+   */
+  void setOp(std::string op) {
```

Comment:
```suggestion
  void setOp(const std::string &op) {
```
How about using const &?

Reply:
done